File: MachineLearning/Part 1 - Data Preprocessing/Data_Preprocessing_Template.py
###### Data Preprocessing Template ########

# Importing the libraries
import numpy as np    
import matplotlib.pyplot as plt        
import pandas as pd

# Importing the Dataset   
dataset = pd.read_csv('Data.csv')
    
# Creating Matrix of Features // Dependent Variable 
X = dataset.iloc[:, :-1].values 
        
# Creating Dependent Variable Vector
Y = dataset.iloc[:, 3].values

# Splitting a dataset into a Training Set and Test Set
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)

# Feature Scaling 
"""
from sklearn.preprocessing import StandardScaler 
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
"""
    

File: MachineLearning/Part 1 - Data Preprocessing/Data_Preprocessing_Template.R
###### Data Preprocessing Template ########

## Importing the Dataset
dataset = read.csv('Data.csv')
# dataset = dataset[, 2:3]

## Splitting a dataset into a Training Set and Test Set
library(caTools)

# To split the dataset, we must first set it to the same seed
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)

# Create Training Set
training_set = subset(dataset, split == TRUE)

# Create Test Set
test_set = subset(dataset, split == FALSE)

## Feature Scaling 
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

#####################################################################################################################
# NOTE: using Function F1 will allow for lookup 
# NOTE: To install packages, simply use 'install.packages('PACKAGE_NAME')'
# NOTE: To run an instance of a package, use 'library(LIBRARY_NAME)
